### ISSUE TITLE
Is it possible to earn achievements using this client?

### ISSUE THREAD

**Issue by awsms** [20PercentRendered/opensteamclient#26]

**Comment by Rosentti** [comment]
Yes. No popups will occur (no overlay support), but you'll get achievements on your profile.

**Comment by awsms** [comment]
Very cool! Do you have an IRC/matrix/discord chan for this project? Feels a bit awkward asking one-liner questions on each issue

**Comment by Rosentti** [comment]
I plan to eventually make a Discord server once the C# version is done.

**Comment by nPHYN1T3** [comment]
There is a reason there is a "Discussions" section rather than "Issues." Discord is exclusionary and keeps info away from people causing the same questions over and over requiring someone there who isn't fed up of answering them.

**Comment by Rosentti** [comment]
> Here is another question, do you have any plan in the future to port Steam's controllers integration? It makes any controller work in basically every single game, using their own database for the key mapping. (I'm talking about game-based controller mapping, not the Steam Controller).

This is supported by IClientController I believe. If you have pre-existing mappings copying over the correct config files should make the bindings work. Otherwise it should also just work, as it's something also handled by `steamclient.so`, but we don't have a UI for editing controller things yet. On Windows though, things will probably become difficult as steam to my knowledge seems to use drivers to implement some functionality.

I could play Spelunky 2 (Proton game) just fine with a DualSense on Linux using the Qt6 OpenSteamClient.

> There is a reason there is a "Discussions" section rather than "Issues." (Project Admins just need to enable it in the Settings.) Discord is exclusionary and keeps info away from people causing the same questions over and over requiring someone there who isn't fed up of answering them. Asking and Answering in the Discussions means the info is always available for visitors (with and without accounts) and indexing.

Ah, if only people actually used Discussions. I'm fine with using the Issues tab for this, since it seems to be higher ranked on Google search results. 
Also having most of the stuff on GitHub is bad for non-programmers, since they don't tend to check it. I also only tend to search the Issues tab instead of Discussions, which would just sit there collecting dust. Sure I could setup issue templates to point to Discussions for some things, but that's a lot of work when I could just make a Discord server and be done with it.

A discord server is good for the average Joe, and also allows me to easily set up FAQs and whatnot. 
I don't really like Discord for them being the massive company they are, but come on. Who games on PC and doesn't have a Discord account? 

But yes, I'll probably make a Discord server, and also file issues on behalf of people who don't have a GitHub account, so things show up on Google, and then make a FAQ (Github Wiki pages don't seem to be indexed in Google search results, so I'm not sure how to do that, maybe as a pinned issue)

**Comment by nPHYN1T3** [comment]
"Who games on PC and doesn't have a Discord account?" - If everyone jumped off a bridge. ;p

Non-Programmers go where the search results for a problem take them. ;)

While I agree about the Discord being a "Massive Company" it's more the fact everything is closed off. It can't be searched meaning someone constantly has to be there to field questions, often the same stuff over and over and over and over when a single post on a static and open page would do the trick...well for those that read. There will always be those FAQing FAQ/Wiki dodgers.

Not everyone uses "Google search results" plus those go through the filter bubble so they aren't reliable person to person anyhow.

**Comment by awsms** [comment]
Indeed that's a massive issue with all these closed platforms, where all the info is gatekept, forces you to create an account, and thus not crawlable by search engines. However you CAN search for any string of text on Discord, so the "you will have to repeat yourself" is not an argument.
Gathering all the info written in the README.md + adding content about the features that the program offers in a wiki sounds like a good idea.

**Comment by awsms** [comment]
It seems like achievements can't be earned using the C# remake

**Comment by Rosentti** [comment]
Interesting, they should definitely be working, unless there's some sort of server connectivity issues. I've also noticed that ValveSteam retrieves userstats quite aggressively all the time, so it's possible I'm missing some API calls?

**Comment by awsms** [comment]
I just re-tried with a random Factorio achievement, and it indeed did not unlock on my Steam profile. Nothing achievement related in the terminal, so some calls are probably missing.
